MobiFlight/Joysticks/Joystick.cs

```diff
@@ -216,14 +233,43 @@ public List<ListItem> GetAvailableDevices()
 
         protected virtual List<JoystickDevice> GetButtonsSorted()
         {
+            var buttons = Buttons.ToArray().ToList();
+            buttons.Sort(SortByPositionInDefintion);
             return Buttons;
         }
 
         protected virtual List<JoystickDevice> GetAxisSorted()
         {
+            var axes = Axes.ToArray().ToList();
+            Axes.Sort(SortByPositionInDefintion);
+
             return Axes;
         }
 
+        public int GetIndexForKey(string key)
+        {
+            return definition?.Inputs?.FindIndex(input => input.Name == key) ?? 0;
+        }
```

Comment:
Updated to use the definition file direction instead of having to maintain a separate `Labels` dictionary.